File: CSharp/07. 계좌예수금조회.cs
namespace CSharp;

internal class _07 : SampleBase
{
    public override async Task ActionImplement()
    {
        // [요청] CSPAQ22200 : 현물계좌예수금 주문가능금액 총평가2
        CSPAQ22200 tr_data = new()
        {
            CSPAQ22200InBlock1 = new(0, "", "", "", "0"),
        };
        await api.GetTRData(tr_data);
        if (tr_data.CSPAQ22200OutBlock1 is null)
        {
            // 오류 처리
            print(tr_data.rsp_cd.Length > 0 ? $"{tr_data.rsp_cd}-{tr_data.rsp_msg}" : api.LastErrorMessage);
            return;
        }

        // tr_data.CSPAQ22200OutBlock 데이터 처리
        print(tr_data.CSPAQ22200OutBlock1);
        print(tr_data.CSPAQ22200OutBlock2);
    }
}

// Output:
/*
CSPAQ22200OutBlock1, Field Count = 5
| Key       | Value       |
|-----------|-------------|
| RecCnt    | 1           |
| MgmtBrnNo |             |
| AcntNo    | XXXXXXXXXXX |
| Pwd       | ********    |
| BalCreTp  | 0           |

CSPAQ22200OutBlock2, Field Count = 37
| Key                    | Value       |
|------------------------|-------------|
| RecCnt                 | 1           |
| BrnNm                  | XXXXXXXXXXX |
| AcntNm                 | XXXXXX      |
| MnyOrdAbleAmt          | 0           |
| SubstOrdAbleAmt        | 0           |
| SeOrdAbleAmt           | 0           |
| KdqOrdAbleAmt          | 0           |
| CrdtPldgOrdAmt         | 0           |
| MgnRat100pctOrdAbleAmt | 0           |
| MgnRat35ordAbleAmt     | 0           |
| MgnRat50ordAbleAmt     | 0           |
| CrdtOrdAbleAmt         | 0           |
| Dps                    | 0           |
| SubstAmt               | 0           |
|
...
*/

File: Python/07. 계좌예수금조회.py
import asyncio
import ebest
from common import *
from app_keys import appkey, appsecretkey # app_keys.py 파일에 appkey, appsecretkey 변수를 정의하고 사용하세요

async def sample(api):
    request = {
        'CSPAQ22200InBlock1': {
            'BalCreTp': '0',
        }
    }
    response = await api.request('CSPAQ22200', request)
    if not response: return print(f'요청실패: {api.last_message}')
    
    print_table(response.body['CSPAQ22200OutBlock1'])
    print_table(response.body['CSPAQ22200OutBlock2'])
    

async def main():
    api=ebest.OpenApi()
    if not await api.login(appkey, appsecretkey):
        return print(f'연결실패: {api.last_message}')
    await sample(api)
    await api.close()

if __name__ == '__main__':
    asyncio.run(main())

# Output:
'''
Field Count = 5
+-----------+-------------+
|    key    |    value    |
+-----------+-------------+
|   RecCnt  |      1      |
| MgmtBrnNo |             |
|   AcntNo  | XXXXXXXXXXX |
|    Pwd    |   ********  |
|  BalCreTp |      0      |
+-----------+-------------+
Field Count = 37
+------------------------+-------------+
|          key           |    value    |
+------------------------+-------------+
|         RecCnt         |      1      |
|         BrnNm          | XXXXXXXXXXX |
|         AcntNm         |    XXXXX    |
|     MnyOrdAbleAmt      |      0      |
|    SubstOrdAbleAmt     |      0      |
|      SeOrdAbleAmt      |      0      |
...
'''